build-info-extractor-go/src/main/java/org/jfrog/build/extractor/go/extractor/GoZipBallStreamer.java

```diff
@@ -238,29 +244,59 @@ private boolean isSubFolderOfAnotherModule(String moduleRootDir, String director
      * @return True if the entry is a go.mod file which doesn't belong to the project we are packing
      */
     private boolean isSubModule(String entryName) {
-        if (entryName.endsWith(MOD_FILE)) {
+        if (entryName.endsWith(MOD_FILE_PATH)) {
             if (shouldPackSubModule()) {
-                return (!entryName.substring(entryName.indexOf('/') + 1).endsWith(subModuleName + MOD_FILE));
+                return (!entryName.substring(entryName.indexOf('/') + 1).endsWith(subModuleName + MOD_FILE_PATH));
             } else {
-                return entryName.substring(entryName.indexOf('/') + 1).endsWith(subModuleName + MOD_FILE);
+                return !entryName.substring(entryName.indexOf('/') + 1).equals(MOD_FILE);
+            }
+        }
+        return false;
+    }
+
+    /**
+     * @param majorVersion - major version of a compatible v2 project
+     * @return true, if the go.mod file in the root specify a path of submodule of the given major version
+     */
+    private boolean hasRootModFileOfCompatibleModuleFromV2(String majorVersion) {
+        Enumeration<? extends ZipArchiveEntry> entries = zipFile.getEntries();
+        if (entries.hasMoreElements()) {
+            ZipArchiveEntry zipEntry = entries.nextElement();
+            ZipArchiveEntry modEntry = zipFile.getEntry(getFirstPathElement(zipEntry.getName()) + MOD_FILE_PATH);
+            if (modEntry == null) {
+                modEntry = zipFile.getEntry(MOD_FILE);
+            }
+            if (modEntry != null) {
+                try (InputStream inputStream = zipFile.getInputStream(modEntry)) {
+                    String modFileContent = IOUtils.toString(inputStream, Charset.defaultCharset());
+                    return modFileContent.substring(0, modFileContent.indexOf("\n")).endsWith("/" + majorVersion);
+                } catch (IOException e) {
+                    log.debug("Failed to read go.mod file of the root project");
```

Comment:
Let's add more information about the failed entry, so we can fix bugs. The information should include at least the full path in the archive to the entry.
Also, please consider changing the `debug` log level to `info` or `warning`, because this is an unexpected error.

Reply:
The full path is the go.mod for the root project (like i mentioned).
I can put it as info, and also add log.debug with the exception error. is it ok?

Reply:
I'm not sure that the `debug` log level is enough. If this exception is unexpected we should print it to the user, so he can report it to us. 
You can add it as an error log message:
```java
log.error("Failed to read go.mod file of the root project", e);
```
Another option is to print the root cause message:
```java
log.warn("Failed to read go.mod file of the root project: " + ExceptionUtils.getRootCauseMessage(e));
```